app/src/main/java/com/example/paginacionexample/viewmodel/MainViewModel.kt

```diff
@@ -11,11 +11,15 @@ class MainViewModel(repository: MainRepository) : ViewModel() {
     lateinit var response: MutableLiveData<ArrayList<MainData>>
 
     var mainDataSource: MainDataSource = repository.requestMainDataSource()
+    var page = 0
 
-    fun getData(page: Int, limit: Int) {
+    init {
         response = mainDataSource.respuesta
-        mainDataSource.requestGetImages(page, limit)
 
     }
 
+    fun getData() {
+        mainDataSource.requestGetImages(page++, 12)
```

Comment:
the 12 could be constant.... but not at this level, maybe in a datasoruce level don't you think ?

Reply:
I believe the same, I converted to constant , and move that to in datasource